core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3328,15 +3341,23 @@ object Stream extends StreamLowPriority {
     * As a result, the returned stream emits no elements and hence has output type `INothing`.
     *
     * Alias for `eval(fa).drain`.
+    */
+  @deprecated("Use evalAction if passing an F[Unit] or eval(fa).drain if passing an F[A]", "2.5.0")
+  def eval_[F[_], A](fa: F[A]): Stream[F, INothing] =
+    new Stream(Eval(fa).map(_ => ()))
+
+  /**
+    * Creates a stream that evaluates the supplied action for its effect and then discards the returned unit.
+    * As a result, the returned stream emits no elements and hence has output type `INothing`.
     *
     * @example {{{
     * scala> import cats.effect.IO
-    * scala> Stream.eval_(IO(println("Ran"))).covaryOutput[Int].compile.toVector.unsafeRunSync
+    * scala> Stream.evalAction(IO(println("Ran"))).covaryOutput[Int].compile.toVector.unsafeRunSync
     * res0: Vector[Int] = Vector()
     * }}}
     */
-  def eval_[F[_], A](fa: F[A]): Stream[F, INothing] =
-    new Stream(Eval(fa).map(_ => ()))
+  def evalAction[F[_]](action: F[Unit]): Stream[F, INothing] =
```

Comment:
do we need another name? I think that for a lot of code this is actually longer than doing `.drain` after, and that still achieves the underlying purpose of making it a bit clearer than things are getting discarded.

You could actually argue that `eval_` is more inline with that since the `_` is "weird" enough to signal something is happening, whereas `evalAction` is a very innocuous name. 

I could definitely picture a lot of `Stream.evalAction(fa).flatMap` in the will.

Reply:
I have found method names that end in underscore, and are named almost as other methods, are a bad idea. They are easy to miss, since most typefaces make underscores barely visible).  Furthermore, in this specific case I would avoid using the word _eval_, or _evaluate_, for an operation (in stream) that returns no value. 

I would suggest using some short words like `do` or `act` to name this operation.

Reply:
@diesalbla `Stream.exec`?